lib/valueflow.cpp

```diff
@@ -4399,7 +4399,7 @@ static void valueFlowLifetime(TokenList *tokenlist, SymbolDatabase*, ErrorLogger
                 else if (c == LifetimeCapture::ByValue)
                     value.lifetimeScope = ValueFlow::Value::LifetimeScope::ThisValue;
                 value.tokvalue = tok2;
-                value.errorPath.push_back({tok2, "Lambda captures the 'this' variable here."});
+                value.errorPath.emplace_back(tok2, "Lambda captures the 'this' variable here.");
```

Comment:
I would have expected for clang-tidy to detect this but maybe that check is still in review.

Reply:
Turns out we have `modernize-use-emplace` disabled in our `.clang-tidy` config.

The detection of unnecessary temporary object was to clang-tidy just today: https://github.com/llvm/llvm-project/commit/987f9cb6b9708ff16980c4f8c3f8ba4674463fb5